src/esperanza/finalizationstate.cpp

```diff
@@ -160,16 +162,15 @@ void FinalizationState::InstaJustify() {
 
 void FinalizationState::IncrementDynasty() {
   // finalized epoch is m_current_epoch - 2 because:
-  // finalized (0) - justified (1) - votes to justify (2)
+  // justified(0) - finalized(1) - votes to justify(2) - m_current_epoch(3)
```

Comment:
justified before finalized?

Reply:
we always justify the previous epoch but if we have 2 consecutive then yes, we put a flag on the last one.

Reply:
sure but the moment that you mark something as finalized everything coming before is considered finalized as well. Having `justified -> finalized` is from a definition point wrong, and anyway super confusing to me.

Reply:
I'll correct the comment

Reply:
It all depends on whether you're reading left to right or right to left?